### ISSUE TITLE
As a user, I want to search for library items by bibliographic metadata so that I can find specific items I’m interested in.

### ISSUE THREAD

**Issue by rlskoeser** [Princeton-CDH/mep-django#226]
# testing notes

make sure you're logged in, then head to the [book search page](https://shakespeareandco.princeton.edu/books/). you should see a shiny new keyword search box like the one on members. type into the box and you should see results load as you type.

**note** - if at any point you notice something funky with the result loading, please comment on #233, since that issue deals with the fast loading behavior specifically. everything else you can keep on this issue.

go ahead and search for a couple works by title and take a look at the relevance scores of your results (make sure you are sorting by relevance). do they make sense? the actual numbers don't mean anything, but the relative _size_ of the numbers should give you an idea of which works solr thinks are most relevant for your query.

please also test the following cases:
- searching for a work by the author's name
- searching for a work by the translator or editor's name
- searching for periodicals by typing "periodical"
- searching for a work by some text in its public notes
- searching for a work that has an author's name in the title (I think there are some books like "James Joyce's Dublin" or similar)

## dev notes

Search should include these fields in this priority order:
* author
* title
* creator (all creator names in a single field)
* format
* public notes

author, title, and creator should probably all be available in both stemmed and unstemmed search fields.

Refer to [PUL Solr config](https://github.com/pulibrary/pul_solr/blob/master/solr_configs/catalog-production/conf/solrconfig.xml#L264-L303) for possible boosting values and unstem search field definition.

**Comment by rlskoeser** [comment]
Marking as blocked because this should wait until after we've integrated OCLC data and decided which fields should be included and how they should be prioritized (= boosted).

**Comment by rlskoeser** [comment]
@clmahoney would you add this item to a future meeting agenda? To implement search on books we need to know which book data fields should be searchable and which ones are most important.

**Comment by clmahoney** [comment]
This looks so awesome! Some notes from testing: 
- Searching seems to work well. I typed in "Moby" and got Moby-Dick right away, with large numbers. 
- If I search for "Homer," I get The Odyssey as the third result. The first two have "Homer" in the title

**Comment by clmahoney** [comment]
- If I search for editors and translators the results vary depending on how much work the editors/translators do. So: if I search for the relatively unknown "Richard Thirsk" then I get his translation of "Ditte" as the 14th result. The previous 13 all have "Richard" in the title or the author's name is Richard. But if I search for "Thirsk" that only shows the Ditte translation. If I search for Eugene Jolas, editor of "transition," that is the 5th result; the previous 4 are all by Jolas also as author. That makes sense.

**Comment by clmahoney** [comment]
- NO: if I search for "periodical" or "periodicals," I do NOT get periodical results, I get this: 
![Screen Shot 2020-03-11 at 3 55 33 PM](https://user-images.githubusercontent.com/16673934/76460103-cbefaf00-63b3-11ea-8126-a59ed8bbd6b6.png)

**Comment by clmahoney** [comment]
- YES, if I search for a work by it's public notes I can find that. For example, I searched for "by or about John Donne" and I got "John Donne" (in quotes) as the first result, which is where the notes belong.

**Comment by clmahoney** [comment]
- YES, if I type in an author's name then a work with that name in the title shows up first. For example: I tried "James Joyce": the first 11 results have James Joyce in the title. The next 9 have a "James" of some kind in the title, usually Henry or William. James Joyce as author does not appear until the 21st entry. I think this might be too far down: is there a way to make it be the next series of results after JJ in the title?

**Comment by thatbudakguy** [comment]
I made some changes to the way books are indexed and boosted, so this is ready to test again.

**Comment by clmahoney** [comment]
- YES, if I search for a work by an author's name, the works by that other appear first. I tried with "Homer" again, and got "The Odyssey" as the first result, then 2 results with "Homer" in the title. I tried with "Melville" and got the first results as books by Melville, then a book called "Herman Melville" in the title. 
- YES, if I search for a work by the translator/editor's name I find the work right away: tried with Thirsk 
- YES! If I search for "periodical" then I see a list of periodicals!! 
![Screen Shot 2020-03-17 at 3 47 42 PM](https://user-images.githubusercontent.com/16673934/76895595-b02e5200-6866-11ea-8f6f-563589073cf7.png)
- YES, if I search by a work by some text in its public notes, I see it
- YES! If I search for a work that has an author's name in the title I find it. I tried with James Joyce, and now the first results are books by James Joyce, then books ABOUT Joyce - great work!!
